utils/wasm-builder/src/stack_end.rs

```diff
@@ -99,11 +81,323 @@ pub fn insert_stack_end_export(
+        let set_global_function_index = (own_module
+            .functions_space()
+            .checked_sub(1)
+            .expect("Must be at least one function already"))
+            as u32;
+
+        log::trace!(
+            "make get/set global functions {} and {} for global {}",
+            get_global_function_index,
+            set_global_function_index,
+            index
+        );
+
+        // Bypass all instructions in module and replace global.get and global.set
+        // by corresponding functions call.
+        for instr in own_module
+            .code_section_mut()
+            .ok_or("Cannot find code section")?
+            .bodies_mut()
+            .iter_mut()
+            .flat_map(|body| body.code_mut().elements_mut().iter_mut())
+        {
+            let global_index = u32::try_from(index).expect("Global index bigger than u32");
+            if *instr == Instruction::GetGlobal(global_index) {
+                *instr = Instruction::Call(get_global_function_index);
+            } else if *instr == Instruction::SetGlobal(global_index) {
+                *instr = Instruction::Call(set_global_function_index);
+            }
+        }
+
+        new_data_in_section.extend(data.iter());
+        global_data_offset += data.len() as u32;
+    }
+
+    // Insert new data section for globals initial values
+    own_module = builder::from_module(own_module)
+        .data()
+        .offset(Instruction::I32Const(data_end_offset as i32))
+        .value(new_data_in_section)
+        .build()
+        .build();
+
+    // Update data end global value
+    handle_mut_global_init_data(
+        module
+            .global_section_mut()
+            .expect("Cannot find globals section")
+            .entries_mut()
+            .get_mut(data_end_index as usize)
+            .expect("We have already find data end global earlier"),
+        |c| {
+            log::debug!(
+                "Change data end offset from {:#x} to {:#x}",
+                c,
+                global_data_offset
+            );
+            *c = global_data_offset as i32;
+        },
+        |_| unreachable!("Data end global has i32 value, which has been already checked"),
+    )?;
+
+    *module = own_module;
+
+    Ok(())
+}
+
+fn get_global_index(module_bytes: &[u8], name_predicate: impl Fn(&str) -> bool) -> Option<u32> {
+    use wasmparser::{Name, NameSectionReader, Parser, Payload};
+
+    Parser::new(0)
+        .parse_all(module_bytes)
+        .filter_map(|p| p.ok())
+        .filter_map(|section| match section {
+            Payload::CustomSection(r) if r.name() == "name" => {
+                Some(NameSectionReader::new(r.data(), r.data_offset()))
+            }
+            _ => None,
+        })
+        .flatten()
+        .filter_map(|res| res.ok())
+        .filter_map(|name| match name {
+            Name::Global(m) => Some(m),
+            _ => None,
+        })
```

Comment:
At least this two `filter_map`s could be replaced by one.
```suggestion
        .filter_map(|name| match name {
            Ok(Name::Global(m)) => Some(m),
            _ => None,
        })
```

Reply:
Done